uniqueness/leaderelection.go

```diff
@@ -40,29 +40,29 @@ import (
 	rl "k8s.io/client-go/tools/leaderelection/resourcelock"
 )
 
-type ElectionManager interface {
+type Manager interface {
 	IsLeader() bool
 	Run(ctx context.Context) error
 	Stop()
 }
 
-type Manager struct {
+type LeaderelectionManager struct {
 	log        *logp.Logger
 	leader     *le.LeaderElector
 	wg         *sync.WaitGroup
 	cancelFunc context.CancelFunc
 	kubeClient k8s.Interface
 }
 
-func NewLeaderElector(log *logp.Logger, cfg *config.Config) ElectionManager {
-	kubeClient, err := providers.KubernetesProvider{}.GetClient(log, cfg.KubeConfig, kubernetes.KubeClientOptions{})
+func NewLeaderElector(log *logp.Logger, cfg *config.Config, k8sProvider providers.KubernetesClientGetter) Manager {
```

Comment:
rename to `NewManager` and also should move to `default_manager.go`

Reply:
IMO, It should remain in `leaderelection.go`, each implementation of the `Manager` interface will create its own `New` function.